Tests/Rules/ReviewUnusedParameter.tests.ps1

```diff
@@ -0,0 +1,68 @@
+Describe "ReviewUnusedParameter" {
+    BeforeAll {
+        $RuleName = 'PSReviewUnusedParameter'
+        $RuleSeverity = "Warning"
+    }
+
+    Context "When there are violations" {
+        $Cases = @(
+            @{
+                ScriptDefinition   = 'function BadFunc1 { param ($Param1, $Param2) $Param1}'
+                Name               = "function with 1 unused parameter"
+                NumberOfViolations = 1
+            }
+            @{
+                ScriptDefinition   = 'function BadFunc1 { param ($Param1, $Param2) }'
+                Name               = "function with 2 unused parameters"
+                NumberOfViolations = 2
+            }
+            @{
+                ScriptDefinition   = '{ param ($Param1) }'
+                Name               = "scriptblock with unused parameter"
+                NumberOfViolations = 1
+            }
+            @{
+                ScriptDefinition   = '{ param ($Param1) }; { $Param1 }'
+                Name               = "parameter with same name in different scope"
+                NumberOfViolations = 1
+            }
+        )
+        It "has 1 violation for <Name>" -TestCases $Cases {
+            param ($ScriptDefinition, $Name, $NumberOfViolations)
+
+            $Violations = Invoke-ScriptAnalyzer -ScriptDefinition $ScriptDefinition -IncludeRule $RuleName
+            $Violations.Count | Should -Be $NumberOfViolations
+            $Violations.Severity | Select-Object -Unique | Should -Be $RuleSeverity
+            $Violations.RuleName | Select-Object -Unique | Should -Be $RuleName
+        }
+    }
+
+    Context "When there are no violations" {
+        $Cases = @(
```

Comment:
Ditto above comment

Reply:
fixed in 938a2835ac8d3c8e5470642d923c3b85476b303c